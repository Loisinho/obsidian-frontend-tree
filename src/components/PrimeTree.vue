<template>
  <div id="primetree">
    <h1>{{ msg }}</h1>
    <p>
      Vue 2 web project using TypeScript as main language,<br>
      PrimeVue components to diplay de data collected with can-ndjson-stream<br>
      and using Jest testing framework.<br>
      <a href="http://download.obsidiansoft.com/frontend.zip">Download this exercise</a>.
    </p>
    <Tree :value="nodes">
      <template #default="nodes">
        {{ nodes.node.name }}
      </template>
    </Tree>
  </div>
</template>

<script lang="ts">
/* eslint-disable @typescript-eslint/no-explicit-any */
import { Component, Prop, Vue } from 'vue-property-decorator';

@Component
export default class PrimeTree extends Vue {
  @Prop() private msg!: string;

  // Tree nodes data.
  nodes: any = [
    {'key': 1, 'name': 'node_1', 'children': [
      {'key': 11, 'name': 'node_1.1', 'children': [
        {'key': 111, 'name': 'node_1.1.1', 'children': []}
      ]},
      {'key': 12, 'name': 'node_1.2', 'children': []}
    ]}
  ];
}
</script>

<style scoped>
#primetree a {
  color: #42b983;
}

#primetree .p-tree {
  text-align: left;
  margin: 50px 10%;
}
</style>
