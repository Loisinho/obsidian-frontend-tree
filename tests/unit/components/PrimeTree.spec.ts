import { mount } from '@vue/test-utils'
import PrimeTree from '@/components/PrimeTree.vue'

let wrapper: any
let spy: jest.SpyInstance<any, unknown[]>

const msg = 'Obsidian Title'
const nodes = [
  {'key': 1, 'id': 1, 'name': 'node_1', 'children': [
    {'key': 11, 'id': 11, 'name': 'node_1.1', 'children': [
      {'key': 111, 'id': 111, 'name': 'node_1.1.1', 'children': []}
    ]},
    {'key': 12, 'id': 12, 'name': 'node_1.2', 'children': []}
  ]}
]

beforeEach(() => {
  wrapper = mount(PrimeTree, {
    propsData: {
      msg
    },
    data() {
      return {
        nodes: nodes
      }
    }
  })

  spy = jest
    .spyOn(wrapper.vm, 'getNodes')
    .mockImplementation(() => nodes)
})

afterEach(() => {
  spy.mockRestore()
})

describe('PrimeTree.vue', () => {
  test('Snapshot', async () => {
    const result = wrapper.vm.getNodes()
    expect(spy).toHaveBeenCalledTimes(1)
    expect(result).toBe(nodes)
    expect(wrapper).toMatchSnapshot()
  })

  test('Tree first node is individually expanded', async () => {
    expect(wrapper.html()).not.toContain('aria-expanded="true"')
    await wrapper.findAll('.p-tree-container > li > div > button').at(0).trigger('click')
    expect(wrapper.text()).toContain(nodes[0].children[0].name)
    expect(wrapper.text()).not.toContain(nodes[0].children[0].children[0].name)
  })
})
