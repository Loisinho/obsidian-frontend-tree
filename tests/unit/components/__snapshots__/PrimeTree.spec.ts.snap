// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimeTree.vue Snapshot 1`] = `
<div id="primetree">
  <h1>Obsidian Title</h1>
  <p>
    Vue 2 web project using TypeScript as main language,<br>
    PrimeVue components to diplay de data collected with can-ndjson-stream<br>
    and using Jest testing framework.<br> <a href="http://download.obsidiansoft.com/frontend.zip">Download this exercise</a>.
  </p>
  <div class="p-tree p-component">
    <!---->
    <!---->
    <ul role="tree" class="p-tree-container">
      <li class="p-treenode">
        <div tabindex="0" role="treeitem" class="p-treenode-content"><button tabindex="-1" class="p-tree-toggler p-link"><span class="p-tree-toggler-icon pi pi-fw pi-chevron-right"></span></button>
          <!----> <span class="p-treenode-icon"></span> <span class="p-treenode-label">
      node_1
    </span>
        </div>
        <!---->
      </li>
    </ul>
  </div>
</div>
`;
